common/achievement/get.ts

```diff
@@ -164,14 +167,12 @@ const assembleAchievementData = async (userAchievements: achievements_with_templ
     let currentValue: number = 0;
     if (currentAchievementTemplate.type === achievement_type_enum.STREAK || currentAchievementTemplate.type === achievement_type_enum.TIERED) {
         const dataAggregationKeys = Object.keys(currentAchievementTemplate.conditionDataAggregations as Prisma.JsonObject);
-        const currentAchievementContext = userAchievements[currentAchievementIndex].context as Prisma.JsonObject;
-        const relation = currentAchievementContext['relation'] as string;
         const evaluationResult = await evaluateAchievement(
             user.userID,
             currentAchievementTemplate.condition,
             currentAchievementTemplate.conditionDataAggregations as ConditionDataAggregations,
             userAchievements[currentAchievementIndex].recordValue || undefined,
-            relation
+            userAchievements[currentAchievementIndex].relation || undefined
```

Comment:
Why || undefined?

Reply:
It's the difference between TS and DB notion. In typescript we are often using `relation?: string` whereas the DB is using `string | null`...

Reply:
I see, and an empty string should also become undefined?

Reply:
Otherwise I'd prefer ?? undefined

Reply:
Also then some care must be taken with Prisma's update, as null means "set to null" whereas undefined means "keep as is" ...

Reply:
👍 It should never be an empty string. Therefore, I would keep `||` in this case.
Besides that I'll make sure that we cannot save an empty string for the creation.